### PR TITLE
feat(app): Improve functionality of ExtraOrders component

### PR DESCRIPTION
**Pull request by ernum**
After some more deliberation, we now allow exhibitors to update their preferences according to the amount chosen.

### CODE REVIEW

File: src/server/api/routers/exhibitor.ts [hunk 1]
```diff
@@ -297,6 +297,22 @@ export const exhibitorRouter = createTRPCRouter({
...
       }
     }),
+  getPreferenceCount: protectedProcedure.query(async ({ ctx }) => {
+    const counts: [{ banqcount: bigint; reprcount: bigint }] =
+      await ctx.prisma.$queryRaw(
+        Prisma.sql`
+        SELECT
+          sum(case when type = 'Banquet' then 1 else 0 end) AS BanqCount,
+          sum(case when type = 'Representative' then 1 else 0 end) AS ReprCount
+        FROM food_specifications
```
Comment (mathiasmagnusson) [inline src/server/api/routers/exhibitor.ts:307]: This counts the food specifications across the whole database, not just the exhibitor of the requesting account, right?

File: src/shared/Classes.ts [hunk 2]
```diff
@@ -97,6 +97,26 @@ export class Package {
...
-
-  getPackage(t: Locale, exhibitorPackage: string) {}
 }
+
+export class Extras {
+  extraChairs: number;
+  extraTables: number;
+  extraDrinkCoupons: number;
+  extraRepresentativeSpots: number;
+  totalBanquetTicketsWanted: number
```
Comment (mathiasmagnusson) [inline src/shared/Classes.ts:107]: Would love if there were a semicolon on this line as well.

File: src/shared/Classes.ts [hunk 3]
```diff
@@ -97,6 +97,26 @@ export class Package {
...
+    extraRepresentativeSpots: number,
+    totalBanquetTicketsWanted: number,
+  ) {
+    this.extraChairs = extraChairs;
+    this.extraTables = extraTables;
+    this.extraDrinkCoupons = extraDrinkCoupons;
+    this.extraRepresentativeSpots = extraRepresentativeSpots;
+    this.totalBanquetTicketsWanted = totalBanquetTicketsWanted;
+  }
+}
```
Comment (mathiasmagnusson) [inline src/shared/Classes.ts:122]: And a trailing newline here

File: src/server/api/routers/exhibitor.ts (hunk 1 above)
Comment (ernum) [reply src/server/api/routers/exhibitor.ts:307]: I forgot to add the where clause. Fixed now.

**Pull request merged**